CMakeLists.txt

```diff
@@ -603,7 +603,7 @@ endif()
 # ----------------------------------------------------------------------------
 # In case of Makefiles if the user does not setup CMAKE_BUILD_TYPE, assume it's Release:
 if(CMAKE_GENERATOR MATCHES "Makefiles|Ninja" AND "${CMAKE_BUILD_TYPE}" STREQUAL "")
-  set(CMAKE_BUILD_TYPE Release)
+  set(CMAKE_DEFAULT_BUILD_TYPE Release)
```

Comment:
> CMAKE_DEFAULT_BUILD_TYPE
> New in version 3.17.

https://cmake.org/cmake/help/latest/variable/CMAKE_DEFAULT_BUILD_TYPE.html

1. We must do not break support for older CMake.
2. What is about `try_compile()` checks?

Reply:
> > added Ninja Multi-Config to regexp
> > regexp
> 
> I do not see any changes here in regexp.

i did not understand what changes should have been made for `ninja multi-config` so i updated as much as i was confident about.
If you could direct me on what changes are needed i would be more than happy to do them